nodl_python/nodl/types.py

```diff
@@ -42,27 +47,19 @@ def __init__(self, *, name: str, value_type: str) -> None:
 
     def __eq__(self, other: Any) -> bool:
         return (
-            isinstance(other, type(self)) or isinstance(self, type(other))
-        ) and self.__dict__ == other.__dict__
+            (isinstance(other, type(self)) or isinstance(self, type(other)))
+            and self.name == other.name
+            and self.type == other.type
+            and (getattr(self, 'role', None) == getattr(other, 'role', None))
```

Comment:
This pattern doesn't seem ideal. Would this be better suited in a child class `__eq__` method that builds on the parent `__eq__`?

Reply:
check out d67e3f7